### PR TITLE
Feature grpc version echo api

### PR DESCRIPTION
**Pull request by JayKaku**
#318 @stefanprodan : Added `version` and `echo` grpc apis and restructured folders and changed `package api` to `package http`

### CODE REVIEW

File: pkg/api/grpc/server.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%v", s.config.Port))
+	if err != nil {
+		s.logger.Fatal("failed to listen", zap.Int("port", s.config.Port))
+	}
+
+	srv := grpc.NewServer()
+	server := health.NewServer()
+	
+	// Register grpc apis for refection
+	echo.RegisterEchoServiceServer(srv, &echoServer{})
```
Comment (stefanprodan) [inline pkg/api/grpc/server.go:75]: The other servers ake a config and logger args, should't this do the same?

File: pkg/api/grpc/echo.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	"github.com/stefanprodan/podinfo/pkg/api/grpc/echo"
+)
+
+type echoServer struct {
+	 echo.UnimplementedEchoServiceServer
+}
+
+func (s *echoServer) Echo (ctx context.Context, message *echo.Message) (*echo.Message, error){
+	log.Printf("Received message body from client: %s", message.Body)
```
Comment (stefanprodan) [inline pkg/api/grpc/echo.go:15]: We need to use the server logger instead of default one.

File: tash [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[1mdiff --git a/cmd/podinfo/main.go b/cmd/podinfo/main.go[m
```
Comment (stefanprodan) [inline tash:1]: Remove this please

**Review by stefanprodan — COMMENTED** [review]
@JayKaku I suggest you rewrite the Git history and have a commit for the api/http rename, one for the grpc and another with echo/panic/version.

File: tash (hunk 3 above)
Comment (JayKaku) [reply tash:1]: Sure

File: pkg/api/grpc/echo.go (hunk 2 above)
Comment (JayKaku) [reply pkg/api/grpc/echo.go:15]: Alright

**Comment by JayKaku** [comment]
> @JayKaku I suggest you rewrite the Git history and have a commit for the api/http rename, one for the grpc and another with echo/panic/version.

@stefanprodan we are planning to raise another PR because the git history is entangled

**Comment by stefanprodan** [comment]
Superseded by https://github.com/stefanprodan/podinfo/pull/322

**Pull request closed without merging**